main/src/ca/uwaterloo/flix/language/phase/unification/BoolFormula.scala

```diff
@@ -25,59 +25,89 @@ import scala.collection.immutable.SortedSet
   * A type class for Boolean Formulas.
   */
 trait BoolFormula[F] {
+
+  /**
+    * Returns `true` if `f` represents TRUE.
+    */
+  def isTrue(f: F): Boolean
+
   /**
-    * Returns the True formula.
+    * Returns `true` if `f` represents FALSE.
+    */
+  def isFalse(f: F): Boolean
+
+  /**
+    * Returns a representation of TRUE.
     */
   def mkTrue: F
 
   /**
-    * Returns the False formula.
+    * Returns a representation of FALSE.
     */
   def mkFalse: F
 
   /**
-    * Returns the a variable formula with the given ID.
+    * Returns a representation of the variable with the given `id`.
     */
   def mkVar(id: Int): F
 
   /**
-    * Returns the formula representing ¬f1
+    * Returns a representation of the complement of `f`.
     */
-  def mkNot(f1: F): F
+  def mkNot(f: F): F
 
   /**
-    * Returns the formula representing f1 ∨ f2.
+    * Returns a representation of the disjunction of `f1` and `f2`.
     */
   def mkOr(f1: F, f2: F): F
 
   /**
-    * Returns the formula representing f1 ∧ f2.
+    * Returns a representation of the conjunction of `f1` and `f2`.
     */
   def mkAnd(f1: F, f2: F): F
 
   /**
-    * Returns the formula f such that f == False <=> f1 == f2.
+    * Returns a representation of the formula `f1 == f2`.
```

Comment:
But this is backwards from what you would expect. Because if they are equal then it returns false.

Reply:
What do you mean? I thought:

```
x == y
x <=> y
x => y and y => x
(not x or y) and (y or notx)
```

Reply:
Hmmmm I guess I was confused.